src/Compilers/CSharp/Portable/CodeGen/EmitConversion.cs

```diff
@@ -26,18 +26,45 @@ private void EmitConversionExpression(BoundConversion conversion, bool used)
                     return;
             }
 
+            var operand = conversion.Operand;
+
+            if (conversion.ConversionKind.IsUserDefinedConversion())
+            {
+                EmitSpecialUserDefinedConversion(conversion, used, operand);
+                return;
+            }
+
             if (!used && !conversion.ConversionHasSideEffects())
             {
-                EmitExpression(conversion.Operand, false); // just do expr side effects
+                EmitExpression(operand, false); // just do expr side effects
                 return;
             }
 
-            EmitExpression(conversion.Operand, true);
+            EmitExpression(operand, true);
             EmitConversion(conversion);
 
             EmitPopIfUnused(used);
         }
 
+        private void EmitSpecialUserDefinedConversion(BoundConversion conversion, bool used, BoundExpression operand)
+        {
+            var typeTo = conversion.Type;
+
+            Debug.Assert((operand.Type.IsArray()) &&
+                         this._module.Compilation.IsReadOnlySpanType(typeTo),
+                         "only special kinds of conversions involvling ReadOnlySpan may be handled in emit");
+
+            if (!TryEmitReadonlySpanAsBlobWrapper((NamedTypeSymbol)typeTo, operand, used, inPlace: false))
+            {
+                EmitExpression(operand, used);
+                if (used)
+                {
+                    _builder.EmitOpCode(ILOpCode.Call, 0);
```

Comment:
Perhaps name `stackAdjustment` as you did in other places. (I was not familiar with that API, so name helps)